content/articles/one-way-anova-excel-testing-normality-homogeneity-using-python/index.md

```diff
@@ -0,0 +1,154 @@
+
+>Note: Highlight the numbers in the data set alone.
+
+- You can decide if you want to view it in the same spreadsheet or another spreadsheet.
+
+![result](/engineering-education/Solving-a-one-way-anova-with-Excel-and-testing-for-normality-and-homogeneity-using-python/result.jpg)
+
+In our ANOVA table above, we analysed the sum of squares and other values of the ANOVA. With this, we can solve a one way ANOVA using Microsoft Excel.
+
+#### Analysis toolpak
+The analysis toolpak is an essential add-in for what we are planning to do. To load the add-in, we'll do the following:
+- Click the file button or the Microsoft logo on the top left corner.
+- Select Excel options and scroll down to add-ins.
+- Out of the list, select Analysis Toolpak and click `Go`.
+- Check on Analysis Toolpak and select `Ok`.
+- Return to the Data tab, and you'll notice the data analysis sub-tab.
+
+#### Testing for normality using box plots
+Now we will create a Python file and import pandas, matplotlib and seaborn into our code. Pandas, Matplotlib & Seaborn are Python libraries for data analysis. 
+
+Pandas is used to import a wide variety of data formats, while Matplotlib and Seaborn are visualisation libraries used to generate powerful visualisations. Seaborn is a library that compiles statistical graphics in Python.
+
+It uses the matplotlib library. Next, we'll read our dataset (it will be in CSV file format). Then we will reshape the data frame suitable for the statistical model package and replace the column names. Now we'll generate a boxplot to see the data distribution by treatments.
+
+>Note: When reading from a directory, use double backslash or put `r` at the front of the directory address encased in a quote symbol.
+
+![unicode error](/engineering-education/Solving-a-one-way-anova-with-Excel-and-testing-for-normality-and-homogeneity-using-python/unicode-error.jpg)
+
+Below is an implementation of the above instructions:
+```Python
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+data_set = pd.read_csv(r"C:\Users\DELL\Desktop\Solving-a-one-way-anova-with-Excel-and-testing-for-normality-and-homogeneity-using-python\assignments.csv")
+data_set_melt = pd.melt(data_set.reset_index(), id_vars=['index'], value_vars=['A', 'B', 'C', 'D', 'E'])
+data_set_melt.columns = ['index', 'treatments', 'value']
+ax = sns.boxplot(x='treatments', y='value', data=data_set_melt, color='#99c2a2')
+ax = sns.swarmplot(x="treatments", y="value", data=data_set_melt, color='#7d0013')
+plt.show()
+```
+
+On implementation, our box plot will be shown as below:
+
+![boxplot](/engineering-education/Solving-a-one-way-anova-with-Excel-and-testing-for-normality-and-homogeneity-using-python/boxplot.jpg)
+
+#### Testing for homogeneity using Bartlett's rule
+To do this, we'll import scipy.stats and call bartlett's method on our data set.
```

Comment:
In a paragraph or two, you can begin by explaining to our readers what the Bartlett's rule entail before writing the code.